Extension/src/LanguageServer/configurations.ts

```diff
@@ -562,7 +569,10 @@ export class CppProperties {
                             });
                             settings.update("default.configurationProvider", undefined); // delete the setting
                         }
+                        let savedKnownCompilers: KnownCompiler[] = this.configurationJson.configurations[0].knownCompilers;
+                        this.configurationJson.configurations[0].knownCompilers = undefined;
```

Comment:
For this usage, it leads to identical behavior (see https://stackoverflow.com/questions/14967535/delete-a-x-vs-a-x-undefined ), but I can change it to use delete though, since is a more "complete" deletion than "= undefined".